File: Aran/RNAV/Libs/Panda/Geometry/C#/Part.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ARAN.GeometryClasses
{
	public class Part : MultiPoint
	{
		public override GeometryType GetGeometryType()
		{
			return GeometryType.Part;
		}

		public override object Clone()
		{
			Part prt = new Part();
			prt.Assign(this);
			return (object)prt;
		}
	}
}

File: Aran/RNAV/Libs/Panda/Geometry/cpp/src/Part.cpp
#include "../include/Part.h"
#include "../include/GeometryType.h"    

namespace Panda
{
	int Part::geometryType () const throw ()
	{
		return GeometryType::Part;
	}

	Geometry* Part::clone () const throw (std::bad_cast)
	{
		Part* part = new Part;
        try
        {
    		part->assign (*this);
        }
        catch (std::bad_cast e)
        {
            delete part;
            throw e;
        }
        
		return part;
	}
}
